### PR TITLE
Markdown formatting fixes, fixes #3481.

### LINKED ISSUE #3481: broken markup in users guide documentation
https://www.haskell.org/cabal/users-guide/developing-packages.html#configuration-flags

> Note that this value may be \[overridden in several ways\](installing-packages.html#controlling-flag-assignments“).

https://www.haskell.org/cabal/users-guide/developing-packages.html#conditional-blocks

>     `if `_condition_
>          _property-descriptions-or-conditionals*_

> or

>     `if `_condition_
>          _property-descriptions-or-conditionals*_
>     `else`
>          _property-descriptions-or-conditionals*_

### PR DESCRIPTION
**Pull request by ezyang**
Signed-off-by: Edward Z. Yang <ezyang@cs.stanford.edu>

### CODE REVIEW

**Comment by 23Skidoo** [comment]
LGTM.

File: Cabal/doc/developing-packages.markdown [hunk 1]
```diff
@@ -1651,9 +1651,9 @@ following fields.
 `default:` _boolean_ (default: `True`)
 :   The default value of this flag.
 
-    Note that this value may be [overridden in several
-    ways](installing-packages.html#controlling-flag-assignments"). The
-    rationale for having flags default to True is that users usually
+    Note that this value may be
+    [overridden in several ways](installing-packages.html#controlling-flag-assignments").
```
Comment (23Skidoo) [inline Cabal/doc/developing-packages.markdown]: The double quotation mark looks suspicious.

**Comment by ezyang** [comment]
Oops, fixed in this one.

**Comment by 23Skidoo** [comment]
Merged, thanks!

**Pull request merged**